File: types/src/relay.ts
import { Client } from './client';

export namespace Relay {
  export const createRelayRequestByUID = ({
    uid,
  }: {
    uid: string;
  }): Client.DeviceRequestByUID['device'] => {
    return {
      uid,
      action: 1,
    };
  };
}

File: app_mobile/lib/models/devices/relay.dart
import 'package:flutter/material.dart';

import '../device.dart';

class RelayData {
  static DeviceRequestDevice getDeviceRequest({@required String uid}) {
    return DeviceRequestDevice(
      uid: uid,
      action: "trigger",
      gpio: 1,
    );
  }
}
